core/src/test/scala/org/apache/spark/ExecutorAllocationManagerSuite.scala

```diff
@@ -787,6 +791,23 @@ class ExecutorAllocationManagerSuite
       Map("host2" -> 1, "host3" -> 2, "host4" -> 1, "host5" -> 2))
   }
 
+  test("SPARK-8366: it should be resubmitted later if task fails") {
```

Comment:
when we come up with a good name, we should probably also update the PR & jira title as well